File: android/TodokTodok/src/main/java/todoktodok/backend/member/application/dto/response/MemberResponse.java
package todoktodok.backend.member.application.dto.response;

import todoktodok.backend.member.domain.Member;

public record MemberResponse(
        Long memberId,
        String nickname,
        String profileImage
) {

    public MemberResponse(final Member member) {
        this(
                member.getId(),
                member.getNickname(),
                member.getProfileImage()
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/latest/MemberResponse.kt
package com.team.todoktodok.data.network.response.latest

import com.team.domain.model.member.Nickname
import com.team.domain.model.member.User
import kotlinx.serialization.Serializable

@Serializable
data class MemberResponse(
    val memberId: Long,
    val nickname: String,
    val profileImage: String,
) {
    fun toDomain(): User = User(memberId, Nickname(nickname), profileImage)
}
